realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/AppExt.kt

```diff
@@ -13,28 +13,32 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.realm
+package io.realm.mongodb
 
+import io.realm.BaseRealmExt
 import io.realm.admin.ServerAdmin
+import io.realm.test_clear
+import io.realm.mongodb.sync.test_reset
 
 /**
  * Resets the Realm Application and delete all local state.
  *
  * Trying to access any Sync or Realm App API's after this has been called has undefined
  * behavior.
  */
-fun RealmApp.close() {
+fun App.close() {
     ServerAdmin().deleteAllUsers()
-    this.syncManager.reset()
-    RealmApp.CREATED = false
-    BaseRealm.applicationContext = null // Required for Realm.init() to work
+    this.syncManager.test_reset()
+    this.networkTransport.resetHeaders()
+    App.CREATED = false
+    BaseRealmExt.test_clear()
```

Comment:
What does this do?

Reply:
Works around that `BaseRealm.applicationContext` is package protected. `BaseRealmExt` is a public extension to `BaseRealm`, but in same package as `BaseRealm`, which allows us to clear the context for test purpose. Overall trying to allow tests to be in `io.realm` to ensure that we test public APIs, with as few hooks into protected state as possible.